src/vs/base/common/search.ts

```diff
@@ -21,3 +25,19 @@ export function buildReplaceStringWithCasePreserved(matches: string[] | null, pa
 		return pattern;
 	}
 }
+
+function validateSpecificSpecialCharacter(matches: string[], pattern: string, specialCharacter: string): boolean {
+	const doesConatinSpecialCharacter = matches[0].indexOf(specialCharacter) !== -1 && pattern.indexOf(specialCharacter) !== -1;
+	return doesConatinSpecialCharacter && matches[0].split(specialCharacter).length === pattern.split(specialCharacter).length;
```

Comment:
Typo: doesConatinSpecialCharacter -> doesContainSpecialCharacter

Reply:
Thanks for this , Fixed it [here](https://github.com/microsoft/vscode/pull/79660/)